Modules/GLO/src/ITSTPCMatchingTask.cxx

```diff
@@ -150,42 +134,86 @@ void ITSTPCMatchingTask::endOfCycle()
+        if (ratio) {
+          ratio->Reset();
+        } else {
+          std::string name = eff->GetName();
+          name += "_Hist";
+          ratio = std::make_unique<common::TH1FRatio>(name.c_str(), eff->GetTitle(),
+                                                      eff->GetPassedHistogram()->GetXaxis()->GetNbins(),
+                                                      eff->GetPassedHistogram()->GetXaxis()->GetXmin(),
+                                                      eff->GetPassedHistogram()->GetXaxis()->GetXmax());
+        }
+        ratio->SetBit(TH1::EStatusBits::kNoStats);
+        if (!ratio->getNum()->Add(eff->GetPassedHistogram())) {
+          ILOG(Error) << "Add operation for numerator histogram of " << ratio->GetName() << " failed; efficiency will be skewed" << ENDM;
+        }
+        if (!ratio->getDen()->Add(eff->GetTotalHistogram())) {
+          ILOG(Error) << "Add operation for denominator histogram of " << ratio->GetName() << " failed; efficiency will be skewed" << ENDM;
+        }
+        ratio->GetXaxis()->SetTitle(eff->GetPassedHistogram()->GetXaxis()->GetTitle());
+        ratio->GetYaxis()->SetTitle(eff->GetPassedHistogram()->GetYaxis()->GetTitle());
+        ratio->Sumw2();
+        ratio->setHasBinominalErrors();
+        ratio->update();
+      };
+
+      // Pt
+      makeRatio(mEffPt, mMatchITSTPCQC.getFractionITSTPCmatch(gloqc::MatchITSTPCQC::ITS));
+      getObjectsManager()->startPublishing(mEffPt.get(), PublicationPolicy::Once);
+      getObjectsManager()->setDefaultDrawOptions(mEffPt->GetName(), "logx");
+      if (mDoPtTrending) {
+        mTrendingPt->AddPoint(mNCycle, mEffPt->GetBinContent(mEffPt->FindBin(mTrendingBinPt)));
+      }
+
+      // Eta
+      makeRatio(mEffEta, mMatchITSTPCQC.getFractionITSTPCmatchEta(gloqc::MatchITSTPCQC::ITS));
+      getObjectsManager()->startPublishing(mEffEta.get(), PublicationPolicy::Once);
+
+      // Phi
+      makeRatio(mEffPhi, mMatchITSTPCQC.getFractionITSTPCmatchPhi(gloqc::MatchITSTPCQC::ITS));
+      getObjectsManager()->startPublishing(mEffPhi.get(), PublicationPolicy::Once);
+    }
+
+    if (mDoK0s) {
+      const auto* k0s = (mIsPbPb) ? mMatchITSTPCQC.getHistoK0MassVsPtVsOccPbPb() : mMatchITSTPCQC.getHistoK0MassVsPtVsOccpp();
+      if (!k0s) {
+        ILOG(Fatal) << "Could not retrieve k0s histogram for beam type: " << mIsPbPb << ENDM;
+      }
+
+      mK0sCycle.reset();
+      mK0sCycle.reset(dynamic_cast<TH3F*>(k0s->Clone("mK0sMassVsPtVsOcc_Cycle")));
+      if (!mK0sCycle) {
+        ILOG(Fatal) << "Could not retrieve k0s histogram for current cycle" << ENDM;
       }
-      if (!ratio->getDen()->Add(eff->GetTotalHistogram())) {
-        ILOG(Error) << "Add operation for denominator histogram of " << name << " failed; efficiency will be skewed" << ENDM;
+      if (!mK0sIntegral) {
+        mK0sIntegral.reset(dynamic_cast<TH3F*>(k0s->Clone("mK0sMassVsPtVsOcc_Integral")));
```

Comment:
ROOT sometimes causes memory issues if there are two histograms with the same name in memory. A workaround is to do what you did a few lines before:
```
      mK0sCycle.reset();
      mK0sCycle.reset(dynamic_cast<TH3F*>(k0s->Clone("mK0sMassVsPtVsOcc_Cycle")));
```
you may want to do the same here.

Reply:
but `mK0sIntegral` is only needed to be initialized once, after that I use `Reset()` and `Add(k0s)` to spare some memory churn, so there should never be two histograms with the same name, or am I misunderstanding something?

Reply:
no, then indeed it should be fine